### PR TITLE
New "for loop" features

### PR DESCRIPTION
**Pull request by dijs**
I kept hating the fact that I had to define an old school while loop:

```
i = 0
while i <= 3
   log i
   i += 1
```

So, instead I created another statement which helps with this:

```
for i in 0 to 3
   log i
```

I know it doesn't save a lot of code or anything, but it just seemed more readable and easier to write.

Also, I was having a hard time understanding the grammar and generator (since I am new to these kind of projects) so my code might not be totally correct. Definitely might need some refactoring...

### CODE REVIEW

File: tests/for_statement.kal [hunk 1]
```diff
@@ -54,3 +54,14 @@ describe 'For Statement', ->
...
     out_s.should.equal 'thisisatest'
 
+  it 'should handle defined iteration values', ->
+    i = 0
+    for index in 0 to 4
+      i.should.equal index
+      i += 1
+
+  it 'should handle defined reverse iteration', ->
+    i = 0
```
Comment (cmwelsh) [inline tests/for_statement.kal:64]: I think this should say:

```
i = 4
```

Comment (dijs) [inline tests/for_statement.kal:64]: That sounds right.

Comment (rzimmerman) [inline tests/for_statement.kal:64]: @cmwelsh I agree.

This got fixed in 06691149db649e68a9e4f96f9177cedd0b388da2

Comment (cmwelsh) [inline tests/for_statement.kal:64]: Ah sorry, I missed that! Appreciate your hard work. :)

**Pull request merged**